server/src/main/java/org/elasticsearch/cluster/ShardHeapUsageSupplier.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.action.ActionListener;
+
+import java.util.Map;
+
+public interface ShardHeapUsageSupplier {
+
+    /**
+     * This will be used when there are no heap usage suppliers available
+     */
+    ShardHeapUsageSupplier EMPTY = listener -> listener.onResponse(Map.of());
+
+    /**
+     * Get the heap usage for every node in the cluster
+     *
+     * @param listener The listener which will receive the results
+     */
+    void getClusterHeapUsage(ActionListener<Map<String, ShardHeapUsage>> listener);
```

Comment:
nit: this doesnt seem like a getter, which would normally return the thing it's getting. Perhaps "collectShardHeapUsages"?

Reply:
Good point, addressed in 7b5bf95d37849d3c93cf69b6594c4629057f7130